From f7ced130751accefe68163c51f0de466d036efcb Mon Sep 17 00:00:00 2001
From: Jiawei Huang <jiawei@tigera.io>
Date: Sat, 14 Jan 2023 13:17:26 -0800
Subject: [PATCH 1/1] Update dependencies to reduce CVEs

This changeset updates Elasticseach dependencies for v7.17.7 to reduce
CVEs detected. The ML artifacts change in [1] is reverted to v7.17.6
behavior as downloading DRA artifacts doesn't seem to work well with
custom Elasticseach builds from source.

[1] https://github.com/elastic/elasticsearch/pull/89985
---
 build-tools-internal/version.properties      |   8 +-
 gradle/verification-metadata.xml             | 176 ++++++++++++++++++-
 modules/transport-netty4/build.gradle        |   1 +
 x-pack/plugin/identity-provider/build.gradle |   2 +-
 x-pack/plugin/ml/build.gradle                |  23 +--
 x-pack/plugin/security/build.gradle          |   6 +-
 6 files changed, 187 insertions(+), 29 deletions(-)

diff --git a/build-tools-internal/version.properties b/build-tools-internal/version.properties
index 90c90471322..6bb75e2bb3e 100644
--- a/build-tools-internal/version.properties
+++ b/build-tools-internal/version.properties
@@ -7,9 +7,9 @@ bundled_jdk = 19+36@877d6127e982470ba2a7faa31cc93d04
 # optional dependencies
 spatial4j         = 0.7
 jts               = 1.15.0
-jackson           = 2.10.4
+jackson           = 2.14.1
 snakeyaml         = 1.33
-icu4j             = 62.1
+icu4j             = 68.2
 supercsv          = 2.4.0
 # when updating log4j, please update also docs/java-api/index.asciidoc
 log4j             = 2.17.1
@@ -17,7 +17,7 @@ slf4j             = 1.6.2
 
 jna               = 5.10.0
 
-netty             = 4.1.66.Final
+netty             = 4.1.87.Final
 joda              = 2.10.10
 
 commons_lang3                   = 3.9
@@ -26,7 +26,7 @@ commons_lang3                   = 3.9
 #  - plugins/ingest-attachment (transitive dependency, check the upstream POM)
 #  - distribution/tools/plugin-cli
 #  - x-pack/plugin/security
-bouncycastle=1.64
+bouncycastle=1.70
 # test dependencies
 randomizedrunner  = 2.8.0
 junit             = 4.12
diff --git a/gradle/verification-metadata.xml b/gradle/verification-metadata.xml
index b367cf74f0b..26e6171669e 100644
--- a/gradle/verification-metadata.xml
+++ b/gradle/verification-metadata.xml
@@ -199,6 +199,11 @@
             <sha256 value="5326a6fbcde7cf8817f36c254101cd45f6acea4258518cd3c80ee5b89f4e4b9b" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.core" name="jackson-annotations" version="2.14.1">
+         <artifact name="jackson-annotations-2.14.1.jar">
+            <sha256 value="d255b4b863ff8ec714a8f96fa55c34621d43dbb82b82d3f57476496a4c09e1e7" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.core" name="jackson-core" version="2.10.4">
          <artifact name="jackson-core-2.10.4.jar">
             <sha256 value="564f6e5706096179537114299e6d7492d2c38da182df8d7834a4c9141b078ef3" origin="Generated by Gradle"/>
@@ -219,6 +224,11 @@
             <sha256 value="ab119a8ea3cc69472ebc0e870b849bfbbe536ad57d613dc38453ccd592ca6a3d" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.core" name="jackson-core" version="2.14.1">
+         <artifact name="jackson-core-2.14.1.jar">
+            <sha256 value="0114187e296b34c931c1bf9e5a84152b62bfab7d182f5623f3982dc2da35e526" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.core" name="jackson-databind" version="2.10.4">
          <artifact name="jackson-databind-2.10.4.jar">
             <sha256 value="55312662a420c71508e6159c86aa41c1694c52e89a1b90dc94bcf4358134005e" origin="Generated by Gradle"/>
@@ -239,11 +249,21 @@
             <sha256 value="6444bf08d8cd4629740afc3db1276938f494728deb663ce585c4e91f6b45eb84" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.core" name="jackson-databind" version="2.14.1">
+         <artifact name="jackson-databind-2.14.1.jar">
+            <sha256 value="423a0c806de4b3fa5eb4a28698305e3a3777c731e1bcfa1b2f3a3760c7b6e773" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-cbor" version="2.10.4">
          <artifact name="jackson-dataformat-cbor-2.10.4.jar">
             <sha256 value="5e15ea426db254c2e98454165b2d9f8dffc4be1f2da4aa18eedf7d7c076cdb59" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-cbor" version="2.14.1">
+         <artifact name="jackson-dataformat-cbor-2.14.1.jar">
+            <sha256 value="aa491e4850fbf278d321d42e924e0471aa51666584baf200d1851bb1ce4562c1" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-smile" version="2.10.4">
          <artifact name="jackson-dataformat-smile-2.10.4.jar">
             <sha256 value="21cb36dbe1eb4782edc35d6eb5da82720f92fbfa3c440f6680e50d8ac65750db" origin="Generated by Gradle"/>
@@ -259,21 +279,41 @@
             <sha256 value="32d4ff882ce88d8f0ac87a04710d60bc2ec1f90ad80da53f0d80fd9477879fb4" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-smile" version="2.14.1">
+         <artifact name="jackson-dataformat-smile-2.14.1.jar">
+            <sha256 value="63dacc4cc9288b89ef9d3e664fed687ba296c9f808e674e95884aea387a9e552" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-xml" version="2.10.4">
          <artifact name="jackson-dataformat-xml-2.10.4.jar">
             <sha256 value="bca111429a4766f8d3f721a249eb1d14ca67809396c8626deb3c89a7fa6823ef" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-xml" version="2.14.1">
+         <artifact name="jackson-dataformat-xml-2.14.1.jar">
+            <sha256 value="90fa6199c194cf3bad58073d3913b3e8a812d06a5db16faa554ba10f5d859436" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-yaml" version="2.10.4">
          <artifact name="jackson-dataformat-yaml-2.10.4.jar">
             <sha256 value="47bf4e8ad64e87def3f4f9ab73c7ed5ca9b51c53cb203b622c29b9640825512c" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.dataformat" name="jackson-dataformat-yaml" version="2.14.1">
+         <artifact name="jackson-dataformat-yaml-2.14.1.jar">
+            <sha256 value="9e157e2625ed855ab73af7915e256f6823993a30982f4923c9ca82bb752a0303" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.datatype" name="jackson-datatype-jsr310" version="2.10.4">
          <artifact name="jackson-datatype-jsr310-2.10.4.jar">
             <sha256 value="86ad491d756afad579f2867701cc2979a66db9300292e726407654c295f799ea" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.datatype" name="jackson-datatype-jsr310" version="2.14.1">
+         <artifact name="jackson-datatype-jsr310-2.14.1.jar">
+            <sha256 value="0b653613f51d6e15396d2b65ccfec9539b2ab471c5842f5b11ac4877cbd51f72" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.jaxrs" name="jackson-jaxrs-base" version="2.10.4">
          <artifact name="jackson-jaxrs-base-2.10.4.jar">
             <sha256 value="3ee1100025dfc51f5fe52c94a64ca5cdaf8775c16d94428ac324b528575aae99" origin="Generated by Gradle"/>
@@ -284,6 +324,11 @@
             <sha256 value="98f27188fa2a72ef5d3f85fab6e6ca0e76bde1a58c9396cb1cf91028080435d6" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.jaxrs" name="jackson-jaxrs-base" version="2.14.1">
+         <artifact name="jackson-jaxrs-base-2.14.1.jar">
+            <sha256 value="4301b7f9a27b97528dc24b863ad3745a52c86a1e3b6fe91a248718f2bb8e23f6" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.jaxrs" name="jackson-jaxrs-json-provider" version="2.10.4">
          <artifact name="jackson-jaxrs-json-provider-2.10.4.jar">
             <sha256 value="5a98a9a2916bef19b531f2699e196fd52201c3230057978852ddb822eedc34f3" origin="Generated by Gradle"/>
@@ -294,6 +339,11 @@
             <sha256 value="f0817100df27ded044dc9ac6effdb9961a3c37327c6c9262ed344218db048c7b" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.jaxrs" name="jackson-jaxrs-json-provider" version="2.14.1">
+         <artifact name="jackson-jaxrs-json-provider-2.14.1.jar">
+            <sha256 value="233b2bfb1234dbe2bcf21759e14baf2de7327f1b7af4178cea1a2acd77b4389f" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.module" name="jackson-module-jaxb-annotations" version="2.10.4">
          <artifact name="jackson-module-jaxb-annotations-2.10.4.jar">
             <sha256 value="1600c21d8d4e98cc22d482e57a437c5c529ec5310fb5970b342a3c006e341e3c" origin="Generated by Gradle"/>
@@ -304,6 +354,11 @@
             <sha256 value="994a0a510a35d55a869567807075736597da97e9d36ad1ebaff5e37def5a55d3" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.module" name="jackson-module-jaxb-annotations" version="2.14.1">
+         <artifact name="jackson-module-jaxb-annotations-2.14.1.jar">
+            <sha256 value="a640e81f2747c73464cc18bb461a9c3f442113be69e81115c0230c521e07963b" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.module" name="jackson-module-kotlin" version="2.12.4">
          <artifact name="jackson-module-kotlin-2.12.4.jar">
             <sha256 value="73993e2bd00354a9dbcaef547f827cd8ab8a7ff0a1cbc3c5b72b585899fb6be4" origin="Generated by Gradle"/>
@@ -314,6 +369,11 @@
             <sha256 value="173b35e183756036457bb2e8d4ce26966fbf04da706a30b68ada69f4a2a14853" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.module" name="jackson-module-kotlin" version="2.14.1">
+         <artifact name="jackson-module-kotlin-2.14.1.jar">
+            <sha256 value="8d83e6fe0ee319266dde031126da7c67024151b7c6a56410c3abb065fcd1ee7c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.jackson.module" name="jackson-module-parameter-names" version="2.12.4">
          <artifact name="jackson-module-parameter-names-2.12.4.jar">
             <sha256 value="d1336fe625a5b030fe99227a721e890a559e43efe0ac05a8bcf7a5e5f8bc6432" origin="Generated by Gradle"/>
@@ -324,6 +384,11 @@
             <sha256 value="6c4c82d16460f41982311efbdd6285a5c898beceacd8c913ef9e3a1ed1a1f4e0" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.fasterxml.jackson.module" name="jackson-module-parameter-names" version="2.14.1">
+         <artifact name="jackson-module-parameter-names-2.14.1.jar">
+            <sha256 value="1b4741c1c566aa321e2aa28f305d2e57d9440a540de416d4cf6b0101a9189b21" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.fasterxml.woodstox" name="woodstox-core" version="5.3.0">
          <artifact name="woodstox-core-5.3.0.jar">
             <sha256 value="b2bd29c31fda49a9b28a22b9e5c2b26443bcfa99c1a28eab70ab9c7d349b5002" origin="Generated by Gradle"/>
@@ -684,6 +749,11 @@
             <sha256 value="e5b0962e566003d68bf3ec7d87b60e279f5cc7aa93cbc4056432803046512478" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.ibm.icu" name="icu4j" version="68.2">
+         <artifact name="icu4j-68.2.jar">
+            <sha256 value="9bd7bf869a44ba8aeb0cddd7e6616e88cd4795ba5bfce2230447cb0e185a646c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.jamesmurty.utils" name="java-xmlbuilder" version="0.4">
          <artifact name="java-xmlbuilder-0.4.jar">
             <sha256 value="681e53c4ffd59fa12068803b259e3a83d43f07a47c112e748a187dee179eb31f" origin="Generated by Gradle"/>
@@ -1214,6 +1284,11 @@
             <sha256 value="99af46a08546da9e03cb5cd6e3daac624771bb08663f304d60a988e27da59cef" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-buffer" version="4.1.87.Final">
+         <artifact name="netty-buffer-4.1.87.Final.jar">
+            <sha256 value="4e3abdf4ddb8810f5fb4a2b57c9144407873b359faa1894007673ed1465dd3e8" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec" version="4.1.42.Final">
          <artifact name="netty-codec-4.1.42.Final.jar">
             <sha256 value="e96ced697fb7df589da7c20c995e01f75a9cb246be242bbc4cd3b4af424ff189" origin="Generated by Gradle"/>
@@ -1229,6 +1304,11 @@
             <sha256 value="d852eab012b0f06c94a625f96404e32e8fe829708a84e10ae71b2d8dfab47f9e" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec" version="4.1.87.Final">
+         <artifact name="netty-codec-4.1.87.Final.jar">
+            <sha256 value="c643fe8318b7c7405acec13ed98e8f634f917960b20486295efbc5a690bac694" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-dns" version="4.1.53.Final">
          <artifact name="netty-codec-dns-4.1.53.Final.jar">
             <sha256 value="747098d17592d092afdfd8c746e099c3b739986461ebcfa68d782f4b5b19b8ae" origin="Generated by Gradle"/>
@@ -1244,6 +1324,11 @@
             <sha256 value="b30b35c69bfe39497aeaf2c7a99bcaea68a9d3eaeee403ecb597cd14f5df4205" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-http" version="4.1.87.Final">
+         <artifact name="netty-codec-http-4.1.87.Final.jar">
+            <sha256 value="39368c15595b791dbe26e7639cf2e71b6647bab8838fa5781cf7a80762eac927" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-http2" version="4.1.53.Final">
          <artifact name="netty-codec-http2-4.1.53.Final.jar">
             <sha256 value="6c777f308be5bb53f33925e695dc97f0b88688e2f1937c2c7a3975e977362df4" origin="Generated by Gradle"/>
@@ -1254,6 +1339,11 @@
             <sha256 value="206e8685fc0e75860b348b2925e6a416d1bace4cea82adbb87abcbc8cb91ad2d" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-http2" version="4.1.87.Final">
+         <artifact name="netty-codec-http2-4.1.87.Final.jar">
+            <sha256 value="e5d01b8db385b23103de67d3666c6fa9fa144d42217a3c8266b2a5efe425f093" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-socks" version="4.1.53.Final">
          <artifact name="netty-codec-socks-4.1.53.Final.jar">
             <sha256 value="8de51d13f48da8eece7df43aa00d6d2923a215668526d8888879fa1534e9aafe" origin="Generated by Gradle"/>
@@ -1264,6 +1354,11 @@
             <sha256 value="f81e9c0c9879769fe5fea1fef54405a9efb0562b4b2f144ab8641819ea38d44b" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-socks" version="4.1.87.Final">
+         <artifact name="netty-codec-socks-4.1.87.Final.jar">
+            <sha256 value="41a76a64514f8f8f50ab8527254073521394138e8a96191905e8c143368b2006" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-common" version="4.1.42.Final">
          <artifact name="netty-common-4.1.42.Final.jar">
             <sha256 value="3d0a918d78292eeca02a7bb2188daa4e5053b6e29b71e6308309033e121242b5" origin="Generated by Gradle"/>
@@ -1279,6 +1374,11 @@
             <sha256 value="bf7e66d832e62dd2cbe7802a3d45ece7f8c6de2958e69d85fae0149cb0820459" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-common" version="4.1.87.Final">
+         <artifact name="netty-common-4.1.87.Final.jar">
+            <sha256 value="811443fc1411a9ef7c1f7b02a750e2a6688f1eec0ae7361ed07812a85d982c86" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-handler" version="4.1.42.Final">
          <artifact name="netty-handler-4.1.42.Final.jar">
             <sha256 value="11eda86500c33b9d386719b5419f513fd9c097d13894f25dd0c75b610d636e03" origin="Generated by Gradle"/>
@@ -1294,6 +1394,11 @@
             <sha256 value="b67da9271458afb434478d86be9f2736a5be2981e49eda2429a9962b10684f7f" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-handler" version="4.1.87.Final">
+         <artifact name="netty-handler-4.1.87.Final.jar">
+            <sha256 value="31a20e1de13b06a677a4719fd5aa95635982114a32c1d034739a48cceada2383" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-handler-proxy" version="4.1.53.Final">
          <artifact name="netty-handler-proxy-4.1.53.Final.jar">
             <sha256 value="1f4b5a540f596e45bcde7c53f4c02780aa5e56eea4103eee78b7d104f90c97b1" origin="Generated by Gradle"/>
@@ -1304,6 +1409,11 @@
             <sha256 value="b300d6276cd7f3c35b16ed5f55ec9d6752ccdebac78473e02d607e1c76792247" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-handler-proxy" version="4.1.87.Final">
+         <artifact name="netty-handler-proxy-4.1.87.Final.jar">
+            <sha256 value="65276b2ab687b1b7c1ceabcc941d6e2cce1038f35b581218d7dbf46b16c5b4ac" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-resolver" version="4.1.42.Final">
          <artifact name="netty-resolver-4.1.42.Final.jar">
             <sha256 value="89768242b6b7cce9bd9f5945ad21d1b4bae515c6b1bf03a8af5d1899779cebc9" origin="Generated by Gradle"/>
@@ -1319,6 +1429,11 @@
             <sha256 value="e89040e9a760f13ec6c05175d633254dd4ea603b204f7a442744485b3c46d1eb" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-resolver" version="4.1.87.Final">
+         <artifact name="netty-resolver-4.1.87.Final.jar">
+            <sha256 value="ee11d22d240b4b125edaed23ec7cc35ad9241b1b8fdd50630de20411ec345fc7" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-resolver-dns" version="4.1.53.Final">
          <artifact name="netty-resolver-dns-4.1.53.Final.jar">
             <sha256 value="bb63961902ce4a4724e80efde16a63b06f871a2cb2387eb90aa2c3663f17db90" origin="Generated by Gradle"/>
@@ -1339,6 +1454,11 @@
             <sha256 value="59e0c2f8e55e0c1d5df254226395a865af87a9e11f8b96d2dff92660b1dbcfc1" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-transport" version="4.1.87.Final">
+         <artifact name="netty-transport-4.1.87.Final.jar">
+            <sha256 value="17ed465cf5b1579b72379e47be3ba7fccb09dda8d95d47c8d668118483b08f43" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-transport-native-epoll" version="4.1.42.Final">
          <artifact name="netty-transport-native-epoll-4.1.42.Final.jar">
             <sha256 value="3c7d659b3bd773e0ea9b7517d2d6baffa275a3d2ae8eb4c10cb8f0a7724b11d5" origin="Generated by Gradle"/>
@@ -1359,6 +1479,11 @@
             <sha256 value="e97306961515012c7ca0dda4f1374ee65d9eb913336b1bac44096231236ad1cf" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-transport-native-unix-common" version="4.1.87.Final">
+         <artifact name="netty-transport-native-unix-common-4.1.87.Final.jar">
+            <sha256 value="cc2d013c933c206605da01be39f4c2c1f4586a1b9103a665bd724bdd46d9cd0c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.opencensus" name="opencensus-api" version="0.28.0">
          <artifact name="opencensus-api-0.28.0.jar">
             <sha256 value="0c1723f3f6d3061323845ce8b88b35fdda500812e0a75b8eb5fcc4ad8c871a95" origin="Generated by Gradle"/>
@@ -1602,6 +1727,11 @@
             <sha256 value="0972bbc99437c4163acd09b630e6c77eab4cfab8a9594621c95466c0c6645396" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="net.minidev" name="accessors-smart" version="2.4.8">
+         <artifact name="accessors-smart-2.4.8.jar">
+            <sha256 value="7dd705aa1ac0e030f8ee2624e8e77239ae1eef6ccc2621c0b8c189866ee1c42c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="net.minidev" name="json-smart" version="2.3">
          <artifact name="json-smart-2.3.jar">
             <sha256 value="903f48c8aa4c3f6426440b8d32de89fa1dc23b1169abde25e4e1d068aa67708b" origin="Generated by Gradle"/>
@@ -1612,6 +1742,11 @@
             <sha256 value="64072f56d9dff5040b2acec477c5d5e6bcebfc88c508f12acb26072d07942146" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="net.minidev" name="json-smart" version="2.4.8">
+         <artifact name="json-smart-2.4.8.jar">
+            <sha256 value="174a9ad578b56644e62b3965d8bf94ac3a76e707c6343b8abac9d3671438b4b2" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="net.sf.ehcache" name="ehcache" version="2.10.4">
          <artifact name="ehcache-2.10.4.jar">
             <sha256 value="32efe00173a887743c614aa625588247fa40edaa56ed8a36ca7d0231cad82468" origin="Generated by Gradle"/>
@@ -2741,6 +2876,11 @@
             <sha256 value="2e2ec8fe0cf873979f630ae4d35e7ede3390321279b7a15de9deed3f3430990c" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.apache.santuario" name="xmlsec" version="2.1.8">
+         <artifact name="xmlsec-2.1.8.jar">
+            <sha256 value="75b6f42147bc8e8718a764821439937fee224a8322ac43ff0b2b406edd7f5c2b" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.apache.servicemix.bundles" name="org.apache.servicemix.bundles.antlr" version="2.7.7_5">
          <artifact name="org.apache.servicemix.bundles.antlr-2.7.7_5.jar">
             <sha256 value="3902794d36d9b81da1b7e697f21ed04ccae276cc116eecc640a4cd0fff2691f2" origin="Generated by Gradle"/>
@@ -2804,6 +2944,11 @@
             <sha256 value="9b88abe9e981d17113a081a4b4261f4082ccdf9144a8a8373fc460c586f6b13a" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.bouncycastle" name="bcmail-jdk15on" version="1.70">
+         <artifact name="bcmail-jdk15on-1.70.jar">
+            <sha256 value="ff6cde372bcabca182e40c1cc5d9b1f9eb73370cad286ce362d3747aff15f230" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.bouncycastle" name="bcpg-fips" version="1.0.4">
          <artifact name="bcpg-fips-1.0.4.jar">
             <sha256 value="b73c80be1099c4756c088cb457a82040509b787519af5c72c9c3d1bff357565e" origin="Generated by Gradle"/>
@@ -2824,6 +2969,11 @@
             <sha256 value="84669138b1d99143e2c009024f67824ab8d3edb9b05b7591f5ebfb020a4bda71" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.bouncycastle" name="bcpkix-jdk15on" version="1.70">
+         <artifact name="bcpkix-jdk15on-1.70.jar">
+            <sha256 value="e5b9cb821df57f70b0593358e89c0e8d7266515da9d088af6c646f63d433c07c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.bouncycastle" name="bcprov-jdk15on" version="1.60">
          <artifact name="bcprov-jdk15on-1.60.jar">
             <sha256 value="7f1a0e6badab38666f8467a9a0ee96656b2f8ec8623867ed34f3cdc173b7ee07" origin="Generated by Gradle"/>
@@ -2839,6 +2989,11 @@
             <sha256 value="e469bd39f936999f256002631003ff022a22951da9d5bd9789c7abfa9763a292" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.bouncycastle" name="bcprov-jdk15on" version="1.70">
+         <artifact name="bcprov-jdk15on-1.70.jar">
+            <sha256 value="8f3c20e3e2d565d26f33e8d4857a37d0d7f8ac39b62a7026496fcab1bdac30d4" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.bouncycastle" name="bctls-fips" version="1.0.9">
          <artifact name="bctls-fips-1.0.9.jar">
             <sha256 value="ada8549467dda0017e098f2362bd7f00841244351d1fc6f89b9da5ff25890151" origin="Generated by Gradle"/>
@@ -3214,6 +3369,11 @@
             <sha256 value="25c7644079f803cd39c5a872404719b12af069a9ee4ca140e7a194972041382c" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.jetbrains.kotlin" name="kotlin-reflect" version="1.5.32">
+         <artifact name="kotlin-reflect-1.5.32.jar">
+            <sha256 value="13237fb82d6e49122d897a1de2b17ac9d5c77137d99070760e6d4191ba9cac31" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.jetbrains.kotlin" name="kotlin-stdlib" version="1.5.10">
          <artifact name="kotlin-stdlib-1.5.10.jar">
             <sha256 value="ca87c454cd3f2e60931f1803c59699d510d3b4b959cd7119296fb947581d722d" origin="Generated by Gradle"/>
@@ -3224,6 +3384,11 @@
             <sha256 value="c55608e9eb6df7327e74b21e271d324dc523cef31587b8d6d2393db08d6e000c" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.jetbrains.kotlin" name="kotlin-stdlib" version="1.5.32">
+         <artifact name="kotlin-stdlib-1.5.32.jar">
+            <sha256 value="452d205f54a1f08673de273cb3d0148fbf9cd95275e5b0068fa1b001f48aee0a" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.jetbrains.kotlin" name="kotlin-stdlib-common" version="1.5.10">
          <artifact name="kotlin-stdlib-common-1.5.10.jar">
             <sha256 value="d958ce94beda85f865829fb95012804866db7d5246615fd71a2f5aabca3e7994" origin="Generated by Gradle"/>
@@ -3234,6 +3399,11 @@
             <sha256 value="8eb3ac530a978422e0f8d0bba78ba2c628bec997dc2f1aa4ef8c5b854e3764b8" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.jetbrains.kotlin" name="kotlin-stdlib-common" version="1.5.32">
+         <artifact name="kotlin-stdlib-common-1.5.32.jar">
+            <sha256 value="dc2fb23e280a39c9abde2cbf4c1567b66840f1f1ece6c7bb78d58d38076b5def" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.jetbrains.kotlin" name="kotlin-stdlib-jdk7" version="1.5.10">
          <artifact name="kotlin-stdlib-jdk7-1.5.10.jar">
             <sha256 value="75ed5680aaacfd94b93c3695d8eb8bfa7cf83893d2e46ca9788345c52d393f8a" origin="Generated by Gradle"/>
@@ -3792,7 +3962,11 @@
             <sha256 value="664543f6a054e314ea0e6ff5118c1e2c32a2f1c3f90b138e45064c17ff3962fc" origin="Generated by Gradle"/>
          </artifact>
       </component>
-
+      <component group="org.yaml" name="snakeyaml" version="1.29">
+         <artifact name="snakeyaml-1.29.jar">
+            <sha256 value="89c5f029811b08c878f0b81dbb05e9626624c1fda4087a26871101e499a217ab" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.yaml" name="snakeyaml" version="1.33">
          <artifact name="snakeyaml-1.33.jar">
             <sha256 value="11ff459788f0a2d781f56a4a86d7e69202cebacd0273d5269c4ae9f02f3fd8f0" origin="Generated by Gradle"/>
diff --git a/modules/transport-netty4/build.gradle b/modules/transport-netty4/build.gradle
index c790030b7f7..289abaf8c5d 100644
--- a/modules/transport-netty4/build.gradle
+++ b/modules/transport-netty4/build.gradle
@@ -35,6 +35,7 @@ dependencies {
   api "io.netty:netty-handler:${versions.netty}"
   api "io.netty:netty-resolver:${versions.netty}"
   api "io.netty:netty-transport:${versions.netty}"
+  api "io.netty:netty-transport-native-unix-common:${versions.netty}"
 }
 
 restResources {
diff --git a/x-pack/plugin/identity-provider/build.gradle b/x-pack/plugin/identity-provider/build.gradle
index d302891aa7c..c174cf726d3 100644
--- a/x-pack/plugin/identity-provider/build.gradle
+++ b/x-pack/plugin/identity-provider/build.gradle
@@ -31,7 +31,7 @@ dependencies {
   api "org.opensaml:opensaml-storage-api:3.4.5"
   api "org.opensaml:opensaml-storage-impl:3.4.5"
   api "net.shibboleth.utilities:java-support:7.5.1"
-  api "org.apache.santuario:xmlsec:2.1.4"
+  api "org.apache.santuario:xmlsec:2.1.8"
   api "io.dropwizard.metrics:metrics-core:3.2.2"
   api ("org.cryptacular:cryptacular:1.2.4") {
       exclude group: 'org.bouncycastle'
diff --git a/x-pack/plugin/ml/build.gradle b/x-pack/plugin/ml/build.gradle
index 5078d946c12..709e168ab0a 100644
--- a/x-pack/plugin/ml/build.gradle
+++ b/x-pack/plugin/ml/build.gradle
@@ -1,5 +1,3 @@
-import org.elasticsearch.gradle.VersionProperties
-
 apply plugin: 'elasticsearch.internal-es-plugin'
 apply plugin: 'elasticsearch.internal-cluster-test'
 apply plugin: 'elasticsearch.internal-test-artifact'
@@ -12,8 +10,6 @@ esplugin {
   extendedPlugins = ['x-pack-autoscaling', 'lang-painless']
 }
 
-def localRepo = providers.systemProperty('build.ml_cpp.repo').orNull
-
 repositories {
   exclusiveContent {
     filter {
@@ -22,26 +18,13 @@ repositories {
     forRepository {
       ivy {
         name "ml-cpp"
+        url providers.systemProperty('build.ml_cpp.repo').orElse('https://prelert-artifacts.s3.amazonaws.com').get()
         metadataSources {
           // no repository metadata, look directly for the artifact
           artifact()
         }
-        if (localRepo) {
-          url localRepo
-          patternLayout {
-            artifact "maven/[orgPath]/[module]/[revision]/[module]-[revision](-[classifier]).[ext]"
-          }
-        } else {
-          url "https://artifacts-snapshot.elastic.co/"
-          patternLayout {
-            if (VersionProperties.isElasticsearchSnapshot()) {
-              artifact '/ml-cpp/[revision]/downloads/ml-cpp/[module]-[revision](-[classifier]).[ext]'
-            } else {
-              // When building locally we always use snapshot artifacts even if passing `-Dbuild.snapshot=false`.
-              // Release builds are always done with a local repo.
-              artifact '/ml-cpp/[revision]-SNAPSHOT/downloads/ml-cpp/[module]-[revision]-SNAPSHOT(-[classifier]).[ext]'
-            }
-          }
+        patternLayout {
+          artifact "maven/org/elasticsearch/ml/ml-cpp/[revision]/ml-cpp-[revision].[ext]"
         }
       }
     }
diff --git a/x-pack/plugin/security/build.gradle b/x-pack/plugin/security/build.gradle
index 5b09033449d..efa7e22c333 100644
--- a/x-pack/plugin/security/build.gradle
+++ b/x-pack/plugin/security/build.gradle
@@ -51,7 +51,7 @@ dependencies {
   api "org.opensaml:opensaml-storage-api:3.4.5"
   api "org.opensaml:opensaml-storage-impl:3.4.5"
   api "net.shibboleth.utilities:java-support:7.5.1"
-  api "org.apache.santuario:xmlsec:2.1.4"
+  api "org.apache.santuario:xmlsec:2.1.8"
   api "io.dropwizard.metrics:metrics-core:3.2.2"
   api ("org.cryptacular:cryptacular:1.2.4") {
     exclude group: 'org.bouncycastle'
@@ -71,8 +71,8 @@ dependencies {
   api "com.nimbusds:lang-tag:1.4.4"
   api "com.sun.mail:jakarta.mail:1.6.3"
   api "net.jcip:jcip-annotations:1.0"
-  api "net.minidev:json-smart:2.4.2"
-  api "net.minidev:accessors-smart:2.4.2"
+  api "net.minidev:json-smart:2.4.8"
+  api "net.minidev:accessors-smart:2.4.8"
   api "org.ow2.asm:asm:9.1"
 
   testImplementation "org.elasticsearch:mocksocket:${versions.mocksocket}"
-- 
2.37.2

